src/docfx/lib/html/HtmlWriter.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Docs.Build
+{
+    /// <summary>
+    /// Provides a high-performance API for forward-only, non-cached writing of HTML read from HtmlReader.
+    /// </summary>
+    internal class HtmlWriter
```

Comment:
>HtmlWriter [](start = 19, length = 10)

shall we have unit tests for `HtmlWriter`?

Reply:
yes